core/src/tx.rs

```diff
@@ -250,7 +250,7 @@ impl TransactionValidator {
     ) -> Result<(), TransactionRejectionReason> {
         debug!("Validating wasm");
 
-        wasm::RuntimeBuilder::new()
+        wasm::RuntimeBuilder::<wasm::state::SmartContract>::new()
```

Comment:
What happened here?

Reply:
Now `RuntimeBuilder` is generic over state which will be used in runtime. Here I specify that we need a `Runtime` for *Smart Contract* executio

Reply:
I can make another API for `RuntimeBuilder`. For example something like this:

```rust
struct RuntimeBuilder;

struct RuntimeBuilderWithState<S> {
    // ...
}

impl RuntimeBuilder {
    pub fn smart_contract() -> RuntimeBuilderWithState<state::SmartContract> {
        // ...
    }
}
```

Reply:
I'd add it to another PR.